prow/proxy-common.inc

```diff
@@ -45,3 +45,11 @@ if [[ -n "${GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
     export BAZEL_BUILD_ARGS="${BAZEL_BUILD_ARGS} --google_credentials=${GOOGLE_APPLICATION_CREDENTIALS} --config=remote-clang-libc++ --config=remote-ci --remote_instance_name=${BAZEL_BUILD_RBE_INSTANCE} --jobs=${BAZEL_BUILD_RBE_JOBS}"
   fi
 fi
+
+# Override envoy.
+if [[ "${ENVOY_REPOSITORY:-}" && "${ENVOY_SHA:-}" && "${ENVOY_PREFIX:-}" ]]; then
+  TMP_DIR=$(mktemp -d -t envoy-XXXXXXXXXX)
+  trap 'rm -rf $TMP_DIR' EXIT
+  BAZEL_BUILD_ARGS="--override_repository=envoy=$TMP_DIR/$ENVOY_PREFIX-$ENVOY_SHA $BAZEL_BUILD_ARGS"
```

Comment:
For readability, please append, unless this breaks functionality due to ordering (but I don't believe that's the case):
```
BAZEL_BUILD_ARGS="${BAZEL_BUILD_ARGS} --override_repository=envoy=$TMP_DIR/$ENVOY_PREFIX-$ENVOY_SHA"
```